### PR TITLE
21 continue fine tuning codet5 with r bioinformatics dataset and evaluate effectiveness

### LINKED ISSUE #21: Continue Fine-Tuning CodeT5 with R Bioinformatics Dataset and Evaluate Effectiveness
Continue with the fine-tuning process of the CodeT5 model using the expanded dataset of R bioinformatics code from sources like Bioconductor, tutorials, and DataScience Meta.

Document all the processes about the effectiveness of the process and what can be improved

Acceptance Criteria:
- Complete the next phase of fine-tuning using the updated dataset.
- Document observations, improvements, and areas needing more data or adjustment.

### PR DESCRIPTION
**Pull request by Omp06**
Fixes #21 

**What was changed?**
A script was created to fine-tune the CodeT5 model on our dataset. This script was based on an article where someone has implemented their dataset and fine-tuned the CodeT5 model. This script removes the unnecessary libraries and extra steps that were involved in the fine-tuning process. 

**Why was it changed?**
The original script that was being utilized was taken straight from the CodeT5 Git Hub. This script required many different files to be configured correctly along with outdated dependencies to be installed. We kept facing issues so we figured we would try to find examples of those who have fine-tuned the same model and go based on that. 

**How was it changed?**
The extra dependencies and configuration files have been removed to remove any complexities. The dataset is simply loaded and preprocessed. The model is loaded with Roberta and then trained with the data. This process is much simpler than how they had described it to be on their Github. 

**Screenshots that show the changes (if applicable):**

### CODE REVIEW

File: Fine Tuning/fine_tuning_script.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+
+# Initialize the model
+model = RobertaForSequenceClassification.from_pretrained("Salesforce/codet5-small")
+
+# Set up TrainingArguments
+training_args = TrainingArguments(
+    output_dir='./results',          
+    evaluation_strategy="epoch",     
+    learning_rate=2e-5,
```
Comment (PranaviKolouju) [inline Fine Tuning/fine_tuning_script.py:42]: Just based on the hyperparameter tuning I did for this model, we may need to reduce the learning rate further to ensure convergence.

File: Fine Tuning/fine_tuning_script.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+model = RobertaForSequenceClassification.from_pretrained("Salesforce/codet5-small")
+
+# Set up TrainingArguments
+training_args = TrainingArguments(
+    output_dir='./results',          
+    evaluation_strategy="epoch",     
+    learning_rate=2e-5,              
+    per_device_train_batch_size=8,   
+    per_device_eval_batch_size=8,    
+    num_train_epochs=3,
```
Comment (PranaviKolouju) [inline Fine Tuning/fine_tuning_script.py:45]: We will likely need more than 3 epochs for when we actually start finetuning the model! If I had to make a guess, it would likely take 50 epochs to ensure convergence.

File: Fine Tuning/fine_tuning_script.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    model=model,                         
+    args=training_args,                 
+    train_dataset=dataset,               
+    eval_dataset=dataset,             
+)
+
+trainer.train()
+
+# Save the model after training
+model.save_pretrained('./final_model')
```
Comment (PranaviKolouju) [inline Fine Tuning/fine_tuning_script.py:61]: I see that you are saving the model after training, which is necessary! I would also suggest that we save out the training and validation losses in a log so that we can monitor training incrementally. I think we can also save out checkpoints after each epoch so that we can decide at what point in training we had the best model.

File: Fine Tuning/fine_tuning_script.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+# Initialize the model
+model = RobertaForSequenceClassification.from_pretrained("Salesforce/codet5-small")
+
+# Set up TrainingArguments
+training_args = TrainingArguments(
+    output_dir='./results',          
+    evaluation_strategy="epoch",     
+    learning_rate=2e-5,              
+    per_device_train_batch_size=8,
```
Comment (PranaviKolouju) [inline Fine Tuning/fine_tuning_script.py:43]: We can probably also look into increasing the batch size since we will be operating on a GPU.

**Review by AjithAkuthota23 — APPROVED** [review]
Good work
